mindmeld/components/classifier.py

```diff
@@ -422,13 +420,7 @@ def _get_model_config(loaded_config=None, **kwargs):
                 model_config.pop("params", None)
         return ModelConfig(**model_config)
 
-    def _data_dump_payload(self):
-        return self._model
-
-    def _create_and_dump_payload(self, path):
-        joblib.dump(self._data_dump_payload(), path)
-
-    def dump(self, model_path, incremental_model_path=None):
+    def dump(self, model_path, incremental_model_path=None, metadata=None):
```

Comment:
what is metadata?

Reply:
I would prefer if metadata is a "ModelMetadata" object that can be standardized read/write APIs across all classifiers

Reply:
@vijay120 I'm slightly confused where should this metadata class reside: in `models/` sub-directory or the `components/`.  The reason for my question is that, while creating a metadata instance for dumping, classifiers have to create an instance of the metadata class, fill it up with metadata and pass it to models. But when loading, models have to create an instance and pass it to classifiers layer. Any suggestions here?